source/ada/lsp-ada_documents.adb

```diff
@@ -281,6 +281,12 @@ package body LSP.Ada_Documents is
          Buffer   => LSP.Types.To_UTF_8_Unbounded_String (Item.text));
       Self.URI := Item.uri;
       Self.LAL := LAL;
+
+      --  After creating an analysis unit, populate the lexical env with it:
+      --  we do this to allow Libadalang to do some work in reaction to
+      --  a file being open in the IDE, in the hope to speed up the response
```

Comment:
I'm a bit concerned by this (and the commit message): Why "in the hope of" ? You should be able to test whether this actually completes your goal or not before doing a pull request no ?

Reply:
I did verify manually and it does seem to help.

Reply:
OK. Why not remove "in the hope" completely then ?